### PR TITLE
AutoML: Add 'TablesClient.get_model_evaluation' method.

### PR DESCRIPTION
**Pull request by jonathan1920**

### CODE REVIEW

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 1]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
         )
         self.__log_operation_info("Batch predict", op)
         return op
+
+    def get_model_evaluation(self, model=None, model_display_name=None,
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: will do!

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 2]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
...
+                model_name=None, project=None, region=None):
+        """Displays evaluation. Also Displays graphs of performance.
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: will do!

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 3]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
...
+            recall_values = []
+            false_positive_rate_lst = []
+            confidence_scores = []
+            true_positive_lst = []
+            false_positive_lst = []
+            false_negative_lst = []
+            true_negative_lst = []
+            confusion_matrix = []
+
+            print("Model classification metrics (threshold at 0.5):")
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: it is; I've added in the threshold as an arg, so the user can set it

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 4]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
...
+                    confusion_matrix.append(confidence_metrics_entry.false_positive_count)
+                    confusion_matrix.append(confidence_metrics_entry.false_negative_count)
+                    confusion_matrix.append(confidence_metrics_entry.true_negative_count)
+
+            print(true_positive_lst)
+            print("Model AUPRC: {}".format(classification_metrics.au_prc))
+            print("Model AUROC: {}".format(classification_metrics.au_roc))
+            print("Model log loss: {}".format(classification_metrics.log_loss))
+            print confusion matrix
+            print("      predicted: NO       predicted: YES")
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: there is! I'll incorporate this. The actual classification information (both the predicted and the actual) is necessary for multi class classification. Assuming it's accessible, I'll use it!

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 5]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
...
+                precision_values.append(confidence_metrics_entry.precision)
+                recall_values.append(confidence_metrics_entry.recall)
+                false_positive_rate_lst.append(confidence_metrics_entry.false_positive_rate)
+
+                true_positive_lst.append(confidence_metrics_entry.true_positive_count)
+                false_positive_lst.append(confidence_metrics_entry.false_positive_count)
+                false_negative_lst.append(confidence_metrics_entry.false_negative_count)
+                true_negative_lst.append(confidence_metrics_entry.true_negative_count)
+                if confidence_metrics_entry.confidence_threshold == 0.5:
+                    print(
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: We can; however, the PR curve effectively does this already. The curve that's not plotted is the F1 score. If you want, we can plot this as well.

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 6]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
...
         return op
+
+    def get_model_evaluation(self, model=None, model_display_name=None,
+                model_name=None, project=None, region=None):
+        """Displays evaluation. Also Displays graphs of performance.
+        Adopted largely from the python-docs-sample/tables/automl/automl_tables_model.py file in GoogleCloudPlatform."""
+
+        import numpy as np
+        from sklearn import metrics
+        import matplotlib.pyplot as plt
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: will do!

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 7]
```diff
@@ -2718,3 +2718,120 @@ def batch_predict(
...
+
+        for evaluation in response:
+            if not evaluation.annotation_spec_id:
+                model_evaluation_id = evaluation.name.split("/")[-1]
+                model_evaluation_full_id = evaluation.name
+        model_evaluation = self.client.get_model_evaluation(model_evaluation_full_id)
+        model_information = self.client.get_model(model_full_id)
+        classification_metrics = model_evaluation.classification_evaluation_metrics
+                                 #message described in the model_evaluation.proto
+                                                   #message described in the classification.proto
```
[the remark this answers is not in the document]
Comment (jonathan1920) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: will do!

**Comment by busunkim96** [comment]
@jonathan1920 Please rebase your branch against master so the diff only shows the changes you've made.

* Checkout master and update with upstream changes. 
```
git checkout master
git pull upstream master
```
* Checkout your branch and rebase.
```
git checkout evaluation
git rebase master
```

**Comment by tseaver** [comment]
Please rebase your branch against the current `master` to remove the spurious changes (there should be nothing outside the `automl` directory).

**Comment by jonathan1920** [comment]
Made a new PR to clean up commit history. https://github.com/googleapis/google-cloud-python/pull/9233

**Pull request closed without merging**